### PR TITLE
AMQPP handler adaptation

### PR DESCRIPTION
**Pull request by mykola-kobets-epam**

### CODE REVIEW

File: amqphandler/amqphandler.go [hunk 1]
```diff
@@ -234,6 +234,7 @@ func (handler *AmqpHandler) SendUnitStatus(unitStatus cloudprotocol.UnitStatus)
 	handler.Lock()
 	defer handler.Unlock()
 
+	unitStatus.MessageType = cloudprotocol.UnitStatusMessageType
```
Comment (al1img) [inline amqphandler/amqphandler.go:237]: Empty line before return.

File: amqphandler/amqphandler.go [hunk 2]
```diff
@@ -234,6 +234,7 @@ func (handler *AmqpHandler) SendUnitStatus(unitStatus cloudprotocol.UnitStatus)
...
 	return handler.scheduleMessage(cloudprotocol.UnitStatusMessageType, unitStatus, false)
```
Comment (al1img) [inline amqphandler/amqphandler.go:238]: Looks like now it is not needed to pass message type to `scheduleMessage` function. Also remove message type  from message header in aos_common

File: amqphandler/amqphandler.go [hunk 3]
```diff
@@ -580,30 +592,30 @@ func (handler *AmqpHandler) runReceiver(deliveryChannel <-chan amqp.Delivery, pa
 				return
 			}
 
+			decryptData, err := handler.cryptoContext.DecryptMetadata(delivery.Body)
+			if err != nil {
+				log.Errorf("Can't decrypt message %s", err)
```
Comment (al1img) [inline amqphandler/amqphandler.go:597]: `: %v`

File: amqphandler/amqphandler.go [hunk 4]
```diff
@@ -580,30 +592,30 @@ func (handler *AmqpHandler) runReceiver(deliveryChannel <-chan amqp.Delivery, pa
...
+
+				continue
+			}
+
 			var incomingMsg cloudprotocol.ReceivedMessage
+			if err := json.Unmarshal(decryptData, &incomingMsg); err != nil {
+				log.Errorf("JSON parsing problem %s", err)
```
Comment (al1img) [inline amqphandler/amqphandler.go:604]: `Can't parse message` will sound better

File: amqphandler/amqphandler.go [hunk 5]
```diff
@@ -580,30 +592,30 @@ func (handler *AmqpHandler) runReceiver(deliveryChannel <-chan amqp.Delivery, pa
...
-				log.Warnf("AMQP unsupported message type: %s", incomingMsg.Header.MessageType)
 				continue
 			}
 
-			decodedData := messageTypeFunc()
-
-			log.Infof("AMQP receive message: %s", incomingMsg.Header.MessageType)
+			decodedData, err := handler.unmarshalReceiveData(incomingMsg.Data)
+			if err != nil {
+				log.Errorf("Can't unmarshal incoming message %s", err)
```
Comment (al1img) [inline amqphandler/amqphandler.go:617]: `: %v`

File: amqphandler/amqphandler.go [hunk 6]
```diff
@@ -612,31 +624,44 @@ func (handler *AmqpHandler) runReceiver(deliveryChannel <-chan amqp.Delivery, pa
...
-		return aoserrors.Wrap(err)
+	// unmarshal type name
+	var messageType struct {
+		Type string `json:"messageType"`
 	}
 
-	if err = json.Unmarshal(decryptData, result); err != nil {
-		return aoserrors.Wrap(err)
+	if err := json.Unmarshal(data, &messageType); err != nil {
+		return nil, aoserrors.New("JSON parsing problem")
```
Comment (al1img) [inline amqphandler/amqphandler.go:638]: no need to override err, return what unmarshal returns

File: amqphandler/amqphandler.go (hunk 1 above)
Comment (mykola-kobets-epam) [reply amqphandler/amqphandler.go:237]: done

File: amqphandler/amqphandler.go (hunk 3 above)
Comment (mykola-kobets-epam) [reply amqphandler/amqphandler.go:597]: fixed

File: amqphandler/amqphandler.go (hunk 4 above)
Comment (mykola-kobets-epam) [reply amqphandler/amqphandler.go:604]: changed

File: amqphandler/amqphandler.go (hunk 6 above)
Comment (mykola-kobets-epam) [reply amqphandler/amqphandler.go:638]: fixed

File: amqphandler/amqphandler.go (hunk 5 above)
Comment (mykola-kobets-epam) [reply amqphandler/amqphandler.go:617]: fixed

**Review by mlohvynenko — APPROVED** [review]

File: amqphandler/amqphandler.go [hunk 7]
```diff
@@ -805,3 +785,32 @@ func (handler *AmqpHandler) sendMessage(
 
 	return nil
 }
+
+func isMessageImportant(message interface{}) bool {
```
Comment (al1img) [inline amqphandler/amqphandler.go:789]: What is the benefit having function with switch instead of array?

File: amqphandler/amqphandler_test.go [hunk 8]
```diff
@@ -958,15 +952,20 @@ func TestSendDisconnectMessages(t *testing.T) {
...
+			// unmarshal type name
+			var message struct {
+				Data struct {
+					MessageType string `json:"messageType"`
+				} `json:"data"`
+			}
 
 			if err = json.Unmarshal(delivery.Body, &message); err != nil {
-				t.Errorf("Error parsing message: %v", err)
+				t.Errorf("Can't parse json message:  err=%v", err)
```
Comment (al1img) [inline amqphandler/amqphandler_test.go:963]: just `%v`

File: amqphandler/amqphandler.go (hunk 7 above)
Comment (mykola-kobets-epam) [reply amqphandler/amqphandler.go:789]: I choose among two options:
using type switch &
array with typenames.
But the latter requires quite redundant syntax to make an array,
like :
reflect.TypeOf(cloudprotocol.DesiredStatus{}).String(), reflect.TypeOf(cloudprotocol.StateAcceptance{}).String(),

File: amqphandler/amqphandler_test.go (hunk 8 above)
Comment (mykola-kobets-epam) [reply amqphandler/amqphandler_test.go:963]: done

**Review by mlohvynenko — APPROVED** [review]

**Pull request merged**